osc/osc.go

```diff
@@ -677,37 +666,44 @@ func readBundle(reader *bufio.Reader, start *int, end int) (*Bundle, error) {
 	return bundle, nil
 }
 
-// readMessage from `reader`.
-func readMessage(reader *bufio.Reader, start *int) (*Message, error) {
+var msgs = &Message{}
+
+// readMessage from `buffer`.
+func readMessage(buffer *bytes.Buffer, start *int) (*Message, error) {
 	// First, read the OSC address
-	addr, n, err := readPaddedString(reader)
+	addr, n, err := readPaddedString(buffer)
 	if err != nil {
 		return nil, err
 	}
 	*start += n
 
 	// Read all arguments
-	msg := NewMessage(addr)
-	if err = readArguments(msg, reader, start); err != nil {
+	msgs.Address = addr
+	msgs.Arguments = msgs.Arguments[:0]
```

Comment:
This is another potential race condition, so while it's more efficient, it's not the best.